src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/DependenciesGraphProvider.cs

```diff
@@ -0,0 +1,653 @@
+                {
+                    continue;
+                }
+
+                var projectContext = ProjectContextProvider.GetProjectContext(projectPath);
+                if (projectContext == null)
+                {
+                    continue;
+                }
+
+                var subTreeProvider = await GetSubTreeProviderAsync(graphContext, inputGraphNode, projectPath).ConfigureAwait(false);
+                if (subTreeProvider == null)
+                {
+                    continue;
+                }
+
+                var nodeInfo = inputGraphNode.GetValue<IDependencyNode>(
+                                    DependenciesGraphSchema.DependencyNodeProperty);
+                if (nodeInfo == null)
+                {
+                    continue;
+                }
+
+                // Refetch reference, projectContext may have been changed since the last time CheckChildren was called
+                nodeInfo = subTreeProvider.GetDependencyNode(nodeInfo.Id);
+                if (nodeInfo == null)
+                {
+                    continue;
+                }
+
+                using (var scope = new GraphTransactionScope())
+                {
+                    var nodeChildren = nodeInfo.Children;
+                    foreach (var childNodeToAdd in nodeChildren)
+                    {
+                        // start tracking changes if needed
+                        if (graphContext.TrackChanges)
+                        {
+                            trackChanges = true;
+                        }
+
+                        inputGraphNode.SetValue(DependenciesGraphSchema.DependencyNodeProperty, nodeInfo);
+
+                        AddGraphNode(graphContext, projectPath, inputGraphNode, childNodeToAdd);
+                    }
+
+                    scope.Complete();
+                }
+
+                if (trackChanges && !ExpandedGraphContexts.Contains(graphContext))
+                {
+                    // Remember this graph context in order to track changes.
+                    // When references change, we will adjust children of this graph as necessary
+                    ExpandedGraphContexts.Add(graphContext);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Generates search graph containing nodes matching search criteria in Solution Explorer 
+        /// and attaches it to correct top level node.
+        /// </summary>
+        private async Task SearchAsync(IGraphContext graphContext)
+        {
+            var searchParametersTypeName = typeof(ISolutionSearchParameters).GUID.ToString();
+            var searchParameters = graphContext.GetValue<ISolutionSearchParameters>(searchParametersTypeName);
+            if (searchParameters == null)
+            {
+                return;
+            }
+
+            var searchTerm = searchParameters.SearchQuery.SearchString;
+            if (searchTerm == null)
```

Comment:
we don't control what is passed from solution explorer and want to be safe and don't show exceptions to users. Yes it might hide some bug in Solution Explorer search, but it would be noticeable anyway since search would not work .... ?

---
In reply to: [71809491](https://github.com/dotnet/roslyn-project-system/pull/309#discussion_r71809491) [](ancestors = 71809491)